core/src/main/java/jenkins/fingerprints/GlobalFingerprintConfiguration.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020, Jenkins project contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.fingerprints;
+
+import hudson.DescriptorExtensionList;
+import hudson.Extension;
+import hudson.ExtensionList;
+import jenkins.model.GlobalConfiguration;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+
+@Extension
+public class GlobalFingerprintConfiguration extends GlobalConfiguration {
+
+    private FingerprintStorage fingerprintStorage;
+
+    public GlobalFingerprintConfiguration() {
+        load();
+    }
+
+    public static GlobalFingerprintConfiguration get() {
+        return GlobalConfiguration.all().getInstance(GlobalFingerprintConfiguration.class);
+    }
+
+    public FingerprintStorage getFingerprintStorage() {
+        return fingerprintStorage;
+    }
+
+    @DataBoundSetter
+    public void setFingerprintStorage(FingerprintStorage fingerprintStorage) {
+        this.fingerprintStorage = fingerprintStorage;
+    }
+
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject json) {
+        json = json.getJSONObject("fingerprints");
```

Comment:
This part does not seem right. IIRC usage of section does not lead to JSON groouping IIRC. Likely you would get null in this conversion

Reply:
That seems to be working fine

Reply:
Something's not right here if you have to do this, the configure method should only contain: https://www.jenkins.io/doc/developer/forms/structured-form-submission/#databinding